enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+     
+     ii.  then scaling a deployment for the Ingress Controller CR which has the value of the `eip-allocations`.
+     
+     iii. then creating a service with service type load balancer with annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+          which uses the value from the field `eip-allocations` of ingress controller CR.
+
+#### 2. Set EIP through the Ingress Controller CRD:
+   
+   The admin sets the `eip-allocations` in the Ingress Controller CR. Then same process
+   as per point 1.c.ii and 1.c.iii is followed by the ingress operator. 
+
+#### 3. Update the existing EIP for the IngressController:
+   
+   When the admin changes the eip-allocations field in the Ingress Controller CR, the watch on the
+   `eip-allocations` field by the ingress operator scales a new deployment for the Ingress Controller CR.
+   Then step 1.c.iii is followed by the ingress operator.
+
+   
+### Risks and Mitigations
+
+Providing a right range of EIP as per AWS guidelines is an admin responsibility.
+If they are not provided the router won't get assigned an IP which can cause communication outage.
+As of now we are not able to identify any more risks which can affect other components of the EP.
+
+### Drawbacks
+
+Currently, there are not any drawbacks as such for this EP.
+
+## Open Questions
+
+1. Shall we support updating EIP in the NLB Ingress Controller ?
+
+Yes, we will support updating the EIP in the custom ingress controller.
+
+## Test Plan
+
+- This EP will be covered by unit tests.
+
+- This  EP will also be covered by end to end tests.
+  
+  The end to end test will cover the scenario where user will set an `eip-allocations` field
+  in the Ingress Controller CR and then test if the Ingress Controller
+  with service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+  has value of `eip-allocations` field from the Ingress Controller CR was created.
+
+  The end to end test will also the cover the scenario where user updates the `eip-allocations` field
+  in the Ingress Controller CR and then test if the Ingress Controller
+  with service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+  has value of `eip-allocations` field from the Ingress Controller CR was updated.
+  
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- N/A
+
+### Tech Preview -> GA
+
+- N/A
+
+### Removing a deprecated feature
+
+- N/A
+
+## Upgrade / Downgrade Strategy
+
+During upgrade, user should be able to populate the eip_allocations_field via installer
+
+During downgrade, the EIPs won't be allocated to the ingress controller's load balancer service.
+
+## Version Skew Strategy
+
+N/A
+
+## Operational Aspects of API Extensions
+
+- If the Ingress Operator can't scale an Ingress Controller with `eip-allocations` then
+  there can be an outage as all the communications to the router will fail.
```

Comment:
The update should fail with an error rather than causing the router to fail.

Reply:
Yes, the progressing status will give a message to delete the LB service manually.

Reply:
What does "scale an Ingress Controller" mean?  Are you saying the cluster-admin has an IngressController that specifies valid `eipAllocations`, and then the cluster-admin changes this IngressController's `spec.replicas` value?  Or are you saying "scale" to mean adding or removing an EIP from `eipAllocations`?

Reply:
just updated
- If Ingress Controller CR is provided with incorrect `eipAllocations` or if the number of `eipAllocations` does
  not match the number of public subnets for VPC ID tagged with cluster name then the Kubernetes LoadBalancer service will be in "Pending" as an LB won't be created in AWS. The reasons for the failure can be found in the status of the  Ingress Controller.